src/core/engines/full-margin-v2/FullMarginEngineV2.sol

```diff
@@ -289,7 +306,22 @@ contract FullMarginEngineV2 is BaseEngine, IMarginEngine {
                             Internal Functions
      ** ========================================================= **/
 
+    function _onlyOwner() internal view {
+        // TODO uncomment when Ownable gets added
+        // if (msg.sender != owner()) revert HV_Unauthorized();
+    }
+
+    /**
+     * @notice gets access status of an address
+     * @dev if whitelist address is not set, it ignores this
+     * @param _address address
+     */
+    function _checkPermissioned(address _address) internal view {
+        if (address(whitelist) != address(0) && !whitelist.grappaAccess(_address)) revert NoAccess();
+    }
+
     function _execute(address _subAccount, ActionArgs[] calldata actions) internal {
+        _checkPermissioned(msg.sender);
```

Comment:
move this to `execute` and `batchExecute` so it don't check several times in `batchExecute` 

also, execute and batch execute can both be external (instead of public)